include/swift/AST/ASTNode.h

```diff
@@ -107,6 +107,19 @@ namespace llvm {
       return LHS.getOpaqueValue() == RHS.getOpaqueValue();
     }
   };
+
+  // A ASTNode is "pointer like".
+  template <>
+  struct PointerLikeTypeTraits<ASTNode> {
+  public:
+    static inline void *getAsVoidPointer(ASTNode N) {
+      return (void *)N.getOpaqueValue();
+    }
+    static inline ASTNode getFromVoidPointer(void *P) {
+      return ASTNode::getFromOpaqueValue(P);
+    }
+    enum { NumLowBitsAvailable = swift::TypeAlignInBits };
```

Comment:
Is this true? I'm pretty sure ASTNode has used up all its low bits

Reply:
Oh, good catch. I just copy-pasted this from somewhere without thinking about it too much. I’ll update it once the stress tester has finished running.